### DISCUSSION TITLE
v-b-toggle in b-card as a stepper doesn't behave as expected

### DISCUSSION THREAD

**Discussion by Darkbat91** [bootstrap-vue/bootstrap-vue#6262] [Q&A] [answered]
Up front i want to say I am not a web developer and am VERY new with Vue. I am just trying to put together a simple walkthrough application for some infrastructure i am building. Everything has gone wonderfully to date with the wealth of information that is available on Vue it has been a pleasure to develop. However, i have ran into an issue i can't seam to resolve on my own, any guidance is appreciated!

## Issue
A collapsible component with `v-b-toggle` gets tied to another component so they open and close together with different IDs

## Expectation
The components would interact separately even back to back

## Troubleshooting
1. Put in same step (everything works as expected)
2. Change from `a` tag to `button` 
3. All 3 different ways of using `v-b-toggle`
4. Add the class `collapsed` specifically to both elements
5. Add an intermediate step between the collapsable steps (issue disappears)
  a. I don't really have anything that needs to go here.
6. Add `v-b-toggle` to the Next button (cant seem to get to work)

## Next thought
I am thinking it might be necessary to pop it out in a modal since for some reason it is not working the way that i am expecting but i really like the flow as it exists if i could just keep this element collapsed.

## Example
A simplified version of the problematic Vue component is below. There is quite a bit more content inside of the collapsed element which is the reason it should be collapsible and not just a hidden password.

```
    "bootstrap": "^4.5.3",
    "bootstrap-vue": "^2.20.1",
```

```
<template>
  <div>
    <b-card v-if="current_step==1" class="card-style" title="Transport">
      <b-card-text class="text-left">
          <ol>
              <li>VPN
                  <a v-b-toggle:vpnpw href="#vpnpw" @click.prevent>show </a>
                  to copy password. <br>
                  <b-collapse id="vpnpw">
                    <code>{{ this.vpnsecret }}</code>
                  </b-collapse>
              </li>
          </ol>
      </b-card-text>
      <b-button class="float-left" variant="secondary" @click="onClickBack">Back</b-button>

      <b-button class="float-right" variant="primary" @click="onClickNext"

       >Next</b-button>

    </b-card>
    <b-card v-if="current_step==2" class="card-style" title="Application">
      <b-card-text class="text-left">
          <ol>
              <li>Application
                 <a v-b-toggle:apppw href="#apppw" @click.prevent> Show</a>
                 to copy password. <br>
                 <b-collapse id="apppw">
                    <code>{{ this.appsecret }}</code>
                 </b-collapse>
              </li>
          </ol>
      </b-card-text>
      <b-button class="float-left" variant="secondary" @click="onClickBack">Back</b-button>
      <b-button class="float-right" variant="primary"
      @click="onClickNext">Next</b-button>
    </b-card>
  </div>
</template>

<script>
export default {
  name: 'Walkthrough',
  data() {
    return {
      current_step: 1,
      max_step: 2,
      appsecret: 'blabla',
      vpnsecret: 'blabla',
    };
  },
  methods: {
    onClickNext() {
      this.current_step += 1;
    },
    onClickBack() {
      this.current_step -= 1;
    },
  },
};
</script>

```

**Comment by Hiws** [comment, marked as answer]
I believe this might be an issue with Vue trying to re-use elements, and not re-rendering correctly when the state is changed.
You can add a `key` to the element with your `v-if` so that Vue knows it needs to re-render everything inside it.

More information about the `key` prop: https://vuejs.org/v2/api/#key

Example (based on the code you provided)
<https://codepen.io/Hiws/pen/poELZgR>

If you believe this is an issue with BootstrapVue, can i get you to open an issue with a codepen for easier debugging (You can fork the one i linked if you want)

**Reply by Darkbat91** [reply]
Thank you so much for the assistance on this, I will have to keep it in mind for future efforts. I do not believe it is an issue, just myself not knowing the proper usage of the framework. I need to read more on the re-use of the elements within Vue.
